apps/web/src/pages/Applications/ApplicationResumeAddPage/components/AddExperienceForm.tsx

```diff
@@ -0,0 +1,206 @@
+import { Button, Heading, Link, Separator } from "@gc-digital-talent/ui";
+import { Select } from "@gc-digital-talent/forms";
+import { errorMessages } from "@gc-digital-talent/i18n";
+import { useAuthorization } from "@gc-digital-talent/auth";
+
+import useRoutes from "~/hooks/useRoutes";
+import { useExperienceMutations } from "~/hooks/useExperienceMutations";
+import { Scalars } from "~/api/generated";
+import { formValuesToSubmitData } from "~/utils/experienceUtils";
+import {
+  ExperienceFormValues,
+  AllExperienceFormValues,
+  ExperienceType,
+} from "~/types/experience";
+import TasksAndResponsibilities from "~/components/ExperienceFormFields/TasksAndResponsibilities";
+import ExperienceDetails from "~/components/ExperienceFormFields/ExperienceDetails";
+
+import { experienceTypeTitles } from "../messages";
+
+type FormAction = "return" | "add-another";
+type ExperienceExperienceFormValues =
+  ExperienceFormValues<AllExperienceFormValues> & {
+    type: ExperienceType | "";
+    action: FormAction | "";
+  };
+export interface AddExperienceFormProps {
+  applicationId: Scalars["ID"];
+}
+
+const AddExperienceForm = ({ applicationId }: AddExperienceFormProps) => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const paths = useRoutes();
+  const { user } = useAuthorization();
+  const methods = useForm<ExperienceExperienceFormValues>();
+  const {
+    watch,
+    register,
+    setValue,
+    formState: { isSubmitSuccessful },
+    reset,
+  } = methods;
+  const [type, action] = watch(["type", "action"]);
+  const { executeMutation, getMutationArgs } = useExperienceMutations(
+    "create",
+    type,
+  );
+  const actionProps = register("action");
+
+  const handleSubmit: SubmitHandler<ExperienceExperienceFormValues> = async (
+    formValues,
+  ) => {
+    const submitData = formValuesToSubmitData(type, formValues, []);
+    const args = getMutationArgs(user?.id || "", submitData);
+    if (executeMutation) {
+      executeMutation(args)
+        .then((res) => {
+          if (res.data) {
+            toast.success(
+              intl.formatMessage({
+                defaultMessage: "Successfully added experience!",
+                id: "DZ775N",
+                description:
+                  "Success message displayed after adding experience to profile",
+              }),
+            );
+            if (formValues.action !== "add-another") {
+              navigate(paths.applicationResume(applicationId));
+            }
+          }
+        })
+        .catch(() => {
+          toast.error(
+            intl.formatMessage({
+              defaultMessage: "Error: adding experience failed",
+              id: "moKAQP",
+              description:
+                "Message displayed to user after experience fails to be created.",
+            }),
+          );
+        });
+    }
+  };
+
+  React.useEffect(() => {
+    if (action === "add-another" && isSubmitSuccessful) {
+      window.scrollTo(0, 0);
```

Comment:
Great idea! I see `react-hook-form` comes with a `setFocus` helper 😱 I removed the scroll to top because the focus scrolls to the input as well and it actually made the experience worse with the `scrollTo` imo.

[focus first input after submit](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6171/commits/aaf0a3fd2e68fa10680292cec80cabee53d0eb4e)